### ISSUE TITLE
Publish the kafka infrastructure to Github registry

### ISSUE THREAD

**Issue by feed3r** [TheG-r-itters/Communications#3]
I did all the prerequisites to create and publish the kafka container to the GitHub container registry, following this guide: 

https://www.codecentric.de/wissens-hub/blog/github-container-registry

I also created a secret and setup this in the repository in Github. Now we need to concretely publish it (point 4 of the guide), but I don't know how to build and publish the image since we have a docker-compose file. 

I know this is really trivial, but don't know how to do.

**Comment by luca-c-xcv** [comment]
The docker compose file is just a sequence of commands that a Docker engine has to follow to create a system composed of multiple containers. So, in the Docker registry, you can’t publish a docker compose file itself but (in this case) the images built with it (or in general, the images built with the command docker build).

To clarify: a Docker container is defined by its Dockerfile. When you build a container (using the rules written in the Dockerfile), you create an image that docker compose (in case you want to create a system of multiple services) uses to create the system required by you. The rules used to create the system of services are written in the docker compose file.

To recap: you have to build an image using `docker compose build` (or `docker build` ). After the images are built, you can publish them to the registry.

**Comment by luca-c-xcv** [comment]
> The docker compose file is just a sequence of commands that a Docker engine has to follow to create a system composed of multiple containers. So, in the Docker registry, you can’t publish a docker compose file itself but (in this case) the images built with it (or in general, the images built with the command docker build).
> 
> To clarify: a Docker container is defined by its Dockerfile. When you build a container (using the rules written in the Dockerfile), you create an image that docker compose (in case you want to create a system of multiple services) uses to create the system required by you. The rules used to create the system of services are written in the docker compose file.
> 
> To recap: you have to build an image using `docker compose build` (or `docker build` ). After the images are built, you can publish them to the registry.

We discussed this topic offline. The main issue was understanding if it’s possible to build an image with a specific tag using the docker compose command. This task is not directly possible; you have to add the tag in the docker-compose rules. However, the best practice is to build the image with docker build and use the references of the published images in the docker-compose file.
